qemu/tests/live_backup_base.py

```diff
@@ -132,7 +132,7 @@ def backup_check(self, compare_image=None):
         for image in self.image_chain:
             params = self.params.object_params(image)
             qemu_image = qemu_storage.QemuImg(params, data_dir, image)
-            qemu_image.check_image(params, data_dir)
+            qemu_image.check_image(params, data_dir, force_share=True)
```

Comment:
Strictly speaking, checking or getting information from a running qcow2 image is incorrect. The reason qcow2 is not allowed is because metadata has to be read from the image file, and it is not safe if the image is being used by the VM, because it may update metadata while we read it, resulting in inconsistent or wrong output. If you really want to check or getting information from a running qcow2 image, the best way is doing that after shutting down the vm.

Reply:
Yes, agree with you. But there are some scenarios that need us to get image info online.
Like, full backup-> space efficiency check-->if ok, then do incremental live backup.

Reply:
Then you should use the Online interface, such as qmp command 'query-block'.
{ "execute": "query-block" }
{"return": [{"io-status": "ok", "device": "drive_image1", "locked": false, "removable": false, "inserted": {"iops_rd": 0, "detect_zeroes": "off", "image": {"virtual-size": 21474836480, "filename": "/home/test/diskfile/mirror2.qcow2", "cluster-size": 65536, "format": "qcow2", "actual-size": 1994244096, "format-specific": {"type": "qcow2", "data": {"compat": "1.1", "lazy-refcounts": false, "refcount-bits": 16, "corrupt": false}}, "dirty-flag": false}, "iops_wr": 0, "ro": false, "node-name": "#block632", "backing_file_depth": 0, "drv": "qcow2", "iops": 0, "bps_wr": 0, "write_threshold": 0, "encrypted": false, "bps": 0, "bps_rd": 0, "cache": {"no-flush": false, "direct": true, "writeback": true}, "file": "/home/test/diskfile/mirror2.qcow2", "encryption_key_missing": false}, "qdev": "image1", "type": "unknown"}]}

Reply:
It's ok for the online image, but for the compared offline backup.img, we can't get its info by { "execute": "query-block" }.

To keep image's safe, we use { "execute": "query-block" } for online image and qemu-img info for offline image, it looks more complicated, so i think it's suitable to use the param "-U"  here.

Reply:
I cannot agree with you. We should run correct steps for different scenarios.